packages/sequelize-transparent-cache/src/cache/util.js

```diff
@@ -54,16 +51,25 @@ function restoreTimestamps (data, instance) {
   })
 }
 
-function generateInclude (model) {
-  return Object.entries(model.associations || [])
-    .filter(([as, association]) => {
-      const hasOptions = Object.prototype.hasOwnProperty.call(association, 'options')
-      return hasOptions
-    })
-    .map(([as, association]) => ({
-      model: model.sequelize.model(association.target.name),
-      as
-    }))
+function generateIncludeRecurse (model, depth = 1) {
```

Comment:
This function is pretty complicated, it would be nice to cover it not only with integration tests, but also add unit tests.

Reply:
Sure. Will do that